homeassistant/components/scheduler/__init__.py

```diff
@@ -0,0 +1,432 @@
+"""Support for scheduling scenes."""
+from datetime import datetime, timedelta
+import logging
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional
+from uuid import uuid4
+
+from dateutil.rrule import rrulestr
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.const import CONF_ENTITY_ID, CONF_ID, MATCH_ALL
+from homeassistant.core import Context, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import (
+    async_track_point_in_time,
+    async_track_state_change,
+)
+import homeassistant.util.dt as dt_util
+from homeassistant.util.json import load_json, save_json
+
+if TYPE_CHECKING:
+    from homeassistant.core import HomeAssistant, ServiceCall, State  # noqa
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = "scheduler"
+
+PERSISTENCE = ".scheduler.json"
+EVENT = "scheduler_updated"
```

Comment:
Why is this needed?

Reply:
I anticipated that someone might build an automation that reacts whenever the scheduler is updated. Since we don't really have that use case yet, I'll remove.